### ISSUE TITLE
Sound not working on production iOS

### ISSUE THREAD

**Issue by josephan** [zmxv/react-native-sound#76]
Hello I am using react-native-sound and when I press the play button on my production iOS app no sound plays. However it works perfectly fine on Android and on the iOS simulator. Unfortunately I don't own an iOS device so I don't know how to start debugging this issue. I only know that the sound doesn't work because users of my app reported it to me.

I am asking for help because I have been stuck on this problem for a few days and I don't know what to do. So any help is gladly appreciated.

Some more info about my application. You can download the iOS app to test it:
[iOS](https://itunes.apple.com/app/id1155917788)
[Android](https://play.google.com/store/apps/details?id=co.joyley.CheeseEnglish)

Dependencies and Version:
```
  "dependencies": {
    "react": "15.3.1",
    "react-native": "0.32.0",
    "react-native-drawer": "^2.3.0",
    "react-native-router-flux": "^3.35.0",
    "react-native-sound": "^0.8.3",
    "react-native-swiper": "^1.4.11",
    "realm": "^0.14.3"
  }
```

How the sound file gets triggered (stripped version of actual code):
```
class Example extends Component {
  constructor(props) {
    super(props)
    this._playSound = this._playSound.bind(this);
  }

  _playSound() {
    var filename = 'sentence.m4a';
    
    var s = new Sound(filename, Sound.MAIN_BUNDLE, (e) => {
      if (e) {
        console.log('error', filename);
      } else {
        this.setState({soundPlaying: true});
        s.play((success) => { this.setState({soundPlaying: false}); });
      }
    })
  }

  render() {
    return (
      <View>
        <TouchableOpacity onPress={this._playSound} />
      </View>
    )
  }
}
```

Here are the exact steps I took to adding the sound files. I have all my sound files located in the folder `/Users/joseph/Music/Converted by MediaHuman`. I dragged and dropped all the sound files into the file navigation sidebar in Xcode 8. To verify that the sound files have been added I checked the Copy Bundle Resources dropdown in Build Phases tab for my project and all the sound files are there. Also I am using the format `.m4a` for my audio files.

**Comment by henninghall** [comment]
I had a similar issue and solved it by replacing "AVAudioSessionCategoryAmbient" with "AVAudioSessionCategoryPlayback" in the enable method in RNSound.m.

**Comment by tannaurus** [comment]
@henninghall that would be the equivalent of setting your sound category to "playback" in the first place, wouldn't it?

**Comment by henninghall** [comment]
@tannaurus I guess so. If I remember correctly, this was written before setCategory was available in the api.

**Comment by poorvasingh04** [comment]
I recently faced this issue and none of the responses available online worked for me. At the end, I found mp3 sounds that I used were not being detected by mac/Xcode. So, when you see file details in Xcode or Finder, it was not able to identify enough information like duration. After spending hours looking for the solution. I found this issue.

What I did wrong - Renamed *.wav or *.avi files to .mp3 manually

Solution - Use iTunes to convert files to specific format. Manually converting file formats corrupts the file and though they work on android, iOS fails to detect them after archiving the build.
